src/main/java/net/snowflake/client/core/HttpUtil.java

```diff
@@ -899,4 +899,18 @@ static int convertSystemPropertyToIntValue(String systemProperty, int defaultVal
     }
     return returnVal;
   }
+
+  /**
+   * Helper function to attach additional headers to a request if present. This takes a (nullable)
+   * map of headers in <name,value> format and adds them to the incoming request using addHeader.
+   *
+   * @param request The request to add headers to. Must not be null.
+   * @param additionalHeaders The headers to add. May be null.
+   */
+  static void applyAdditionalHeaders(
```

Comment:
Can you comment that which applictaion right now using it? e.g. SnowSight and also some background why it's needed.

Reply:
Done. I left a more detailed description in Confluence and linked it here (I saw other places were also having links to Confluence).